### PR TITLE
Prepare proper release

### PR DESCRIPTION
**Pull request by garyb**
@paf31 please take a look at the doc comments and let me know what you think. It took me _far_ too long to come up with some of this stuff!

### CODE REVIEW

File: src/Data/Newtype.purs [hunk 1]
```diff
@@ -17,6 +43,17 @@ ala
...
 
+-- | Similar to `ala` but useful for cases where you want to use an additional
+-- | projection with the higher order function:
+-- |
+-- | ``` purescript
+-- | ala' Additive foldMap String.length ["hello", "world"] -- 10
+-- | ala' Multiplicative foldMap Math.abs [1.0, -2.0, 3.0, -4.0] -- 24.0
+-- | ```
+-- |
+-- | The type admits other possibilities due to the polymorphic `Functor`
```
Comment (paf31) [inline src/Data/Newtype.purs:54]: I think another useful example here is `traverse`, right?

Comment (garyb) [reply src/Data/Newtype.purs:54]: Yeah, I believe so, have you got a short example that would work? I figured it'd probably be useful with `transformers`, but I've not got to the point that builds with all the other stuff yet, so I couldn't test anything out.

**Comment by paf31** [comment]
Looks good. It might be nice to add a function to get an `Iso` for any newtype:

``` purescript
iso :: forall p new old. (Newtype new old, Profunctor p) => p old old -> p new new
iso = dimap wrap unwrap
```

Actually, you can generalize that type a bit, but it might not have good type inference properties.

**Comment by garyb** [comment]
I was thinking this would be pretty high up the dependency hierarchy, since it's useful for `monoid`, etc. so perhaps that'd be better in `profunctor`? I agree though, there's a lot of overlap with iso stuff here.

**Comment by paf31** [comment]
Sounds good.

File: src/Data/Newtype.purs (hunk 1 above)
Comment (garyb) [reply src/Data/Newtype.purs:54]: I'll merge this as it is, can always add more examples later - at least there's some explanation of what the combinators do now!

File: src/Data/Newtype.purs [hunk 2]
```diff
@@ -17,6 +43,17 @@ ala
   -> f a
 ala _ f = map unwrap (f wrap)
 
+-- | Similar to `ala` but useful for cases where you want to use an additional
+-- | projection with the higher order function:
+-- |
+-- | ``` purescript
+-- | ala' Additive foldMap String.length ["hello", "world"] -- 10
```
Comment (paf31) [inline src/Data/Newtype.purs]: `alaF` not `ala'`

Comment (garyb) [reply src/Data/Newtype.purs]: Oops, thanks!

Comment (garyb) [reply src/Data/Newtype.purs]: Actually, I'll move all of the examples into a tests file so they're at least typechecked, etc. with stubs for the "external" functions.

Comment (garyb) [reply src/Data/Newtype.purs]: Actually I won't, it's more trouble than it's worth :smile:

**Pull request merged**